src/rgw/rgw_sal.cc

```diff
@@ -777,6 +778,14 @@ int RGWRadosStore::create_bucket(RGWUser& u, const rgw_bucket& b,
 	return ret;
       }
     }
+
+    if (policy) {
+      if (!(old_policy == *policy)) {
```

Comment:
there's an acl owner comparison above `(old_policy.get_owner().get_id().compare(u.get_id()) != 0)` that should probably happen inside of the `operator==()` instead. it's also probably worth providing both `operator==` and `operator!=` for convenience here